File: conversions/picocolors/elide-picocolors.rb
#!/usr/bin/env ruby

# Ruby Integration Example for elide-picocolors
#
# This demonstrates calling the TypeScript picocolors implementation from Ruby
# using Elide's polyglot capabilities.
#
# Benefits:
# - One picocolors implementation shared across TypeScript and Ruby
# - Consistent terminal coloring behavior across microservices
# - No Ruby library discrepancies
# - Perfect for Rails, Sidekiq, background workers

puts "=== Ruby Calling TypeScript Picocolors ===\n"

# NOTE: Exact syntax depends on Elide's Ruby polyglot API (currently alpha)
# This is a CONCEPTUAL example - adjust when Ruby support is ready

# Assuming Elide provides something like:
# picocolors_module = Elide.require('./elide-picocolors.ts')

# Example 1: Basic Usage
# result = picocolors_module.default(input_data)
# puts "Result: #{result}"
# puts ""

# Example 2: Rails Integration
# class DataService
#   def initialize
#     @picocolors = picocolors_module
#   end
#
#   def process(data)
#     @picocolors.default(data)
#   end
# end
#
# service = DataService.new
# result = service.process(input_data)
# puts "Processed: #{result}"
# puts ""

# Example 3: Sidekiq Worker
# class ProcessingJob
#   include Sidekiq::Worker
#
#   def perform(data)
#     result = picocolors_module.default(data)
#     puts "Job result: #{result}"
#   end
# end

puts "Real-world use case:"
puts "- Ruby worker uses picocolors"
puts "- Uses same TypeScript implementation as Node.js API"
puts "- Guarantees consistent terminal coloring across services"
puts "- Colorize CLI output"
puts ""

puts "Example: Microservices Architecture"
puts "┌─────────────────────────────────────────┐"
puts "│   Elide Picocolors (TypeScript)           │"
puts "│   elide-picocolors.ts                        │"
puts "└─────────────────────────────────────────┘"
puts "         ↓           ↓           ↓"
puts "    ┌────────┐  ┌────────┐  ┌────────┐"
puts "    │ Node.js│  │  Ruby  │  │ Rails  │"
puts "    │  API   │  │ Worker │  │  App   │"
puts "    └────────┘  └────────┘  └────────┘"
puts "         ↓           ↓           ↓"
puts "    Same terminal coloring everywhere!"
puts ""

puts "Problem Solved:"
puts "Before: Ruby + JavaScript = different implementations"
puts "After: One Elide implementation = 100% consistent"
puts ""

puts "Rails Integration (when ready):"
puts "  # config/initializers/picocolors.rb"
puts "  PICOCOLORS = Elide.require('./elide-picocolors.ts')"
puts "  "
puts "  # app/services/data_service.rb"
puts "  class DataService"
puts "    def process(data)"
puts "      PICOCOLORS.default(data)"
puts "    end"
puts "  end"

File: conversions/picocolors/elide-picocolors.py
#!/usr/bin/env python3
"""
Python Integration Example for elide-picocolors

This demonstrates calling the TypeScript picocolors implementation from Python
using Elide's polyglot capabilities.

Benefits:
- One picocolors implementation shared across TypeScript and Python
- Consistent terminal coloring behavior across services
- No Python library inconsistencies
- Perfect for data pipelines and processing
"""

# NOTE: Exact syntax depends on Elide's Python polyglot API (currently alpha)
# This is a CONCEPTUAL example - adjust when Python support is ready

# Assuming Elide provides something like:
# from elide import require
# picocolors_module = require('./elide-picocolors.ts')

print("=== Python Calling TypeScript Picocolors ===\n")

# Example 1: Basic Usage
# result = picocolors_module.default(input_data)
# print(f"Result: {result}")
# print()

# Example 2: Data Pipeline
# class DataProcessor:
#     def __init__(self):
#         self.picocolors = picocolors_module
#
#     def process_batch(self, items):
#         """Process batch using picocolors"""
#         return [self.picocolors.default(item) for item in items]
#
# processor = DataProcessor()
# data = ['item1', 'item2', 'item3']
# results = processor.process_batch(data)
# for i, result in enumerate(results, 1):
#     print(f"  Item {i}: {result}")
# print()

print("Real-world use case:")
print("- Python data pipeline uses picocolors")
print("- Uses same TypeScript implementation as Node.js API")
print("- Guarantees consistent terminal coloring across services")
print("- Colorize CLI output")
print()

print("Example: Cross-Service Architecture")
print("┌─────────────────────────────────────────┐")
print("│   Elide Picocolors (TypeScript)           │")
print("│   elide-picocolors.ts                        │")
print("└─────────────────────────────────────────┘")
print("         ↓                   ↓")
print("    ┌────────┐          ┌────────┐")
print("    │ Node.js│          │ Python │")
print("    │  API   │          │Pipeline│")
print("    └────────┘          └────────┘")
print("         ↓                   ↓")
print("    Same terminal coloring everywhere!")
print()

print("Problem Solved:")
print("Before: Python + JavaScript = different terminal coloring implementations")
print("After: One Elide implementation = 100% consistent behavior")
print()

print("When Elide Python API is ready, usage will be:")
print("  from elide import require")
print("  picocolors = require('./elide-picocolors.ts')")
print("  result = picocolors.default(data)")
